.github/workflows/codspeed.yml

```diff
@@ -41,19 +41,39 @@ jobs:
         if: steps.cache-py.outputs.cache-hit != 'true'
 
       - name: install rust stable
+        id: rust-toolchain
         uses: dtolnay/rust-toolchain@stable
+        with:
+          components: llvm-tools
 
       - name: cache rust
         uses: Swatinem/rust-cache@v2
 
-      - name: Install pydantic-core
+      - name: Compile pydantic-core for profiling
+        # build-prod does not include MiMalloc because it bypasses Maturin
+        run: |
+          pip install -e .
+          make build-prod
+          python -c 'import pydantic_core'
+        env:
+          CONST_RANDOM_SEED: 0 # Fix the compile time RNG seed
+          RUSTFLAGS: "-Cprofile-generate=${{ github.workspace }}/profdata"
+
+      - name: Gather pgo data
+        run: pytest
```

Comment:
Is pytest the best thing to use for pgo data?

I would guess it has more failures than real world data. Maybe it's a good start.

Reply:
I figured exercising as much of the codebase as possible will lead to the best results, let's see what #686 tells us.